docs/analytics-ga4/ga4-finder-tracker.md

```diff
@@ -0,0 +1,122 @@
+
+## Basic use
+
+```html
+<form id="myForm">
+  <div data-ga4-form-container>
+    <div data-ga4-section="Topics">
+        <select data-ga4-change-category="update-filter select">
+            <option value="1">Default option</option>
+            <option value="2">Option 1</option>
+            <option value="3">Option 2</option>
+        </select>
+        <input type="search" data-ga4-change-category="update-keyword text" />
+    </div>
+  </div>
+</form>
+```
+
+```JavaScript
+var myForm = document.querySelector('myForm')
+
+window.GOVUK.analyticsGa4.Ga4FinderTracker.setFilterIndexes()
+
+myForm.addEventListener('change', function(event) {
+    var ga4ChangeCategory = event.target.closest('[data-ga4-change-category]').getAttribute('data-ga4-change-category')
+    window.GOVUK.analyticsGa4.Ga4FinderTracker.trackChangeEvent(event.target, ga4ChangeCategory)
+})
+```
+
+The tracker is not a regular module. Instead it is called manually where needed. This is to allow the tracker to be called at the appropriate time in our finders' code. In our case, we only call `Ga4FinderTracker.trackChangeEvent()` when our search successfully displays new results to the user. This stops erroring change events (i.e. the user inputting non-dates into the date filter) from being tracked as an "added filter" event. Therefore we only track updates to the finders if they result in a successful update to the search results.
+
+The flow of the tracker is:
+
+1. On your finder results page, tag your elements with the appropriate `data` attributes.
+
+    a. Tag the parent `<div>` or wrapper element for your filters with `data-ga4-filter-container`. This tells our code where to look for filter sections, so that we can set an `index_section` for each filter.
+
+    b. Tag each filters' wrapper element with a `data-ga4-section` attribute. This key can also be populated with a string value, which will be used to populate the `section` value of our GTM object. For example, on a "Topics" filter `<div>` you could add `data-ga4-section="Topics"` or just `data-ga4-section`.
+
+    c. Tag each element of the form that can trigger a `change` event with a `data-ga4-change-category`. This will contain some metadata about what the change is. The current categories are: `update-filter`, `update-sort`, `clear-all-filters` and `update-keyword`. As well as this, add the type of element that the filter is. This assists with our tracker extracting the value of the filter. For example, a `<select>` element that updates a filter would have `data-ga4-change-category="update-filter select"`. A search box would have `data-ga4-change-category="update-keyword text"`.
```

Comment:
Do you think there'd be any mileage in listing all the element types?

Reply:
Oh wait, just noticed you go into them in a later section. Maybe we could say "As well as this, add the type of element that the filter is (further details on what types are tracked is documented below)."